core/src/main/java/org/testcontainers/containers/VncRecordingContainer.java

```diff
@@ -83,23 +93,69 @@ protected void configure() {
-                        "flvrec.py -o " + RECORDING_FILE_NAME + " -d -r " + frameRate + " -P /vnc_password " + targetNetworkAlias + " " + vncPort
+                        "flvrec.py -o " + ORIGINAL_RECORDING_FILE_NAME + " -d -r " + frameRate + " -P /vnc_password " + targetNetworkAlias + " " + vncPort
         );
     }
 
+    @SneakyThrows
+    public void saveRecordingToFile(@NonNull File file) {
+        try(InputStream inputStream = streamRecording()) {
+            Files.copy(inputStream, file.toPath(), StandardCopyOption.REPLACE_EXISTING);
+        }
+    }
+
     @SneakyThrows
     public InputStream streamRecording() {
+        String newRecordingFileName = recordingFormat.reencodeRecording(this, ORIGINAL_RECORDING_FILE_NAME);
+
         TarArchiveInputStream archiveInputStream = new TarArchiveInputStream(
-                dockerClient.copyArchiveFromContainerCmd(getContainerId(), RECORDING_FILE_NAME).exec()
+                dockerClient.copyArchiveFromContainerCmd(getContainerId(), newRecordingFileName).exec()
         );
         archiveInputStream.getNextEntry();
         return archiveInputStream;
     }
 
-    @SneakyThrows
-    public void saveRecordingToFile(File file) {
-        try(InputStream inputStream = streamRecording()) {
-            Files.copy(inputStream, file.toPath(), StandardCopyOption.REPLACE_EXISTING);
+
+    public enum VncRecordingFormat {
+        FLV("flv") {
+            @Override
+            String reencodeRecording(@NonNull VncRecordingContainer container, @NonNull String source) throws IOException, InterruptedException {
+                String newFileOutput = "/newScreen.flv";
+                container.execInContainer("ffmpeg" , "-i", source, "-vcodec", "libx264", newFileOutput);
+                return newFileOutput;
+            }
+        },
+        MP4("mp4") {
+            @Override
+            String reencodeRecording(@NonNull VncRecordingContainer container, @NonNull String source) throws IOException, InterruptedException {
+                String newFileOutput = "/newScreen.mp4";
+                container.execInContainer("ffmpeg" , "-i", source, "-vcodec", "libx264", "-movflags", "faststart", newFileOutput);
+                return newFileOutput;
+            }
+        };
+
+        abstract String reencodeRecording(VncRecordingContainer container, String source) throws IOException, InterruptedException;
+
+        private static final VncRecordingFormat DEFAULT_FORMAT = FLV;
+        private final String filenameExtension;
+
+        VncRecordingFormat(String filenameExtension) {
+            this.filenameExtension = filenameExtension;
+        }
+
+        public String getExtension() {
+            return filenameExtension;
+        }
+
+        /**
+         * @return {@code vncRecordingFormat} value if not null, otherwise, {@link VncRecordingFormat#FLV} will be returned as a default format.
+         */
+        public static VncRecordingFormat of(VncRecordingFormat vncRecordingFormat) {
```

Comment:
what's the purpose of this method?

Reply:
Just in case of `null`, so It can fall back to the default format. Indeed I wasn't sure If it was necessary, but just in case, to prevent a possible `NullPointerException`. 
But I get rid of it for now, thanks.